### PR TITLE
Feature/7961 config refactor

### PR DESCRIPTION
**Pull request by adrian-fitsios**
[AB#7961](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7961)
[AB#8385](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8385)

### CODE REVIEW

**Comment by adrian-fitsios** [comment]
Not ready for a review yet, temporarily closing the PR

**Comment by adrian-fitsios** [comment]
I think this is the minimum amount of work we need that will allow us to substitute config values easily. 

Originally I wanted to continue this refactor and move all config related properties out of UITest class and have the other classes use Settings directly. 

What are your thoughts @talawson05 @PhysicsJon ? 

Should I continue with the refactor or are we happy with it as is?

Any feedback regarding the refactor is welcome, obviously.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -86,9 +72,13 @@ steps:
...
-    command: 'test'
-    projects: 'src/NHSDPublicBrowseAcceptanceTests.Tests/*.csproj'
-    arguments: '-v n'
\ No newline at end of file
+- bash: |
+    export DB__SERVERURL=gpitfutures-dev-sql-pri.database.windows.net
+    export DB__DATABASENAME=bc-buyingcatalogue-development-helm-bapi
+    export DB__PASSWORD="$(gpitfuturesdevdbpassword)"
+    export AZUREBLOBSTORAGE__CONNECTIONSTRING="$(gpitdevstorageaccountconnectionstring)"
+    export AZUREBLOBSTORAGE__CONTAINERNAME=buyingcatalogue-development-helm-documents
```
Comment (talawson05) [inline azure-pipelines.yml]: can the server url, db name, and az container name be set as parameters that the can be set at the pipeline? make it easier later to point at different things

File: azure-pipelines.yml [hunk 2]
```diff
@@ -10,8 +10,11 @@ pool:
   vmImage: 'ubuntu-latest'
 
 variables:
-  buildConfiguration: 'Release'
-  BROWSER: chrome
+  - name: buildConfiguration
+    value: 'Release'
+  - name: BROWSER
+    value: chrome
+  - group: dev-secrets
```
Comment (talawson05) [inline azure-pipelines.yml:23]: so it looks like we're pulling the secrets from the dev-secrets thing, is this a pipeline thing or a kubernetes thing and how do i set the values when running locally? just override the appsettings.json?

Comment (adrian-fitsios) [reply azure-pipelines.yml:23]: That's a 'variable group' and you can find it in AzDo under: Pipelines > Library.

When you're running locally, you can either override the appsettings.json (and make sure not to commit those changes) or you can set env vars that will override them. I think the readme has a section on how to do that. Just note that in order to override a value that's nested - the server url for example, you need to name the var like so: PARENTKEY__CHILDKEY - for more examples have a look at the 'runTests' task in azure-pipelines.yaml

**Review by talawson05 — APPROVED** [review]
I think I'm happy, be interested in Jon's thoughts

**Review by PhysicsJon — APPROVED** [review]
I like it, puts us nicely on the road to deployable tests

**Pull request merged**